tests/unit/azure/test_access.py

```diff
@@ -103,47 +122,116 @@ def test_save_spn_permissions_valid_azure_storage_account():
     )
 
 
-def test_save_spn_permissions_no_valid_storage_accounts(caplog, mocker, az_token):
+def test_create_global_spn_no_policy():
+    w = create_autospec(WorkspaceClient)
+    location = ExternalLocations(w, MockBackend, "ucx")
+    installation = create_autospec(Installation)
+    installation.load.return_value = WorkspaceConfig(inventory_database='ucx')
+    azure_resources = create_autospec(AzureResources)
+    azure_resource_permission = AzureResourcePermissions(installation, w, azure_resources, location)
+    with pytest.raises(ValueError):
+        azure_resource_permission.create_uber_principal()
+
+
+def test_create_global_spn_spn_present():
+    w = create_autospec(WorkspaceClient)
+    location = ExternalLocations(w, MockBackend, "ucx")
```

Comment:
```suggestion
    location = ExternalLocations(w, MockBackend(), "ucx")
```
Why type checker missed this bug?